client/resource_source.go

```diff
@@ -201,17 +201,17 @@ func s3aSourceDestinationSchema() *schema.Resource {
 			},
 			"endpoint": {
 				Type:         schema.TypeString,
-				Required:     true,
+				Optional:     true,
 				ValidateFunc: validation.StringIsNotWhiteSpace,
 			},
 			"access_key": {
 				Type:         schema.TypeString,
-				Required:     true,
+				Optional:     true,
 				ValidateFunc: validation.StringIsNotWhiteSpace,
 			},
 			"secret_key": {
 				Type:         schema.TypeString,
-				Required:     true,
+				Optional:     true,
```

Comment:
The bottom two are required together no? It might make sense to add an additional validation.